pkg/rc/replication_controller.go

```diff
@@ -316,6 +318,70 @@ func (rc *replicationController) addPods(current types.PodLocations, eligible []
 	return nil
 }
 
+func (rc *replicationController) allocateNodes(currentNodes []types.NodeName, eligibleNodes []types.NodeName) error {
+	var err error
+	if rc.AllocationStrategy == fields.CattleStrategy {
+		err = rc.allocateCattleNodes(currentNodes, eligibleNodes)
+	} else {
+		err = rc.allocatePetNodes(currentNodes, eligibleNodes)
+	}
+
+	if err != nil {
+		return util.Errorf("Unable to allocate nodes: %s", err)
+	}
+
+	return nil
+}
+
+func (rc *replicationController) allocateCattleNodes(currentNodes []types.NodeName, eligibleNodes []types.NodeName) error {
+	rcStatus, _, err := rc.rcStatusStore.Get(rc.ID())
+	isNoNodeTransferInProgress := (err != nil && statusstore.IsNoStatus(err)) ||
```

Comment:
i think you're not handling the `err !=nil && !statusstore.IsNoStatus(err)` case here (in other words, when there is a real error that prevents you from reading the status.

When there are a lot of cases I like using switch statements. you could do something like this:

```
switch {
case err == nil || statusstore.IsNoStatus(err):
  isNoNodeTransferInProgress = true
default:
  return err
}
```